### PR TITLE
Flush logger context for LR on exit

### PR DESCRIPTION
**Pull request by hisundar**
+Add in cleanshutdown too

## Overview

Description:

Why should this be merged: 

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [ ] There are no TODOs left in the code
- [ ] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [ ] Change is covered by automated tests
- [ ] Public API has Javadoc

### CODE REVIEW

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuInterClusterReplicationServer.java [hunk 1]
```diff
@@ -271,6 +272,9 @@ private void startServer() {
         }
 
         log.info("main: Server exiting due to shutdown");
+        // Flush the async appender before exiting to prevent the loss of logs
```
Comment (annym) [inline infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuInterClusterReplicationServer.java:275]: can we move this to a flushAsyncAppender() function?

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuInterClusterReplicationServer.java [hunk 2]
```diff
@@ -341,6 +345,9 @@ public void cleanShutdown() {
         if (replicationDiscoveryService != null) {
             replicationDiscoveryService.shutdown();
         }
+        // Flush the async appender before exiting to prevent the loss of logs
```
Comment (zhangn49) [inline infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuInterClusterReplicationServer.java:348]: Should we add a check to see if the `activeServer` is null?
If `startDiscoveryService` throws an exception, activeServer will be null and the shutdown hook hit NPE. Logger flush will be skipped.

https://github.com/CorfuDB/CorfuDB/blob/fa011a6f95c1bf586d9f8f22509e312fa3bd52bf/infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuInterClusterReplicationServer.java#L344

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuInterClusterReplicationServer.java (hunk 1 above)
Comment (hisundar) [reply infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuInterClusterReplicationServer.java:275]: sure. done

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuInterClusterReplicationServer.java (hunk 2 above)
Comment (hisundar) [reply infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuInterClusterReplicationServer.java:348]: done. thanks

**Pull request merged**